addons/web_editor/models/field_html_history_diff.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+import re
+
+from datetime import datetime
+from odoo import api, fields, models
+from odoo.tools.misc import _format_time_ago
+from difflib import SequenceMatcher
+
+
+# ------------------------------------------------------------
+# History restoration methods
+# ------------------------------------------------------------
+
+
+def restore_one(current_content, diff_list):
```

Comment:
First of all, all name of variables an function are open to discution as I didn't put a lot of tough into it. So yes we ca talk about it.

It's call `restore_one` because it restore only one "history", a history is one list of diff. 
You can request a restore from version 8 to version 5 for exemple, it will loop over 3 `restore_one`.

As for the documentation, indeed, I though about witing it when the code will be more or less stable, and no more big changes / refactoring was in view.